src/main/java/org/example/application/CountHealthInsuranceContributionService.java

```diff
@@ -0,0 +1,99 @@
+        this.countMonthlyRevenueService = countMonthlyMonthlyRevenueService;
+    }
+
+    @Override
+    public Money calculateHealthInsuranceContribution(Customer customer, YearMonth monthToCalculateHealthInsuranceContribution) {
+
+        if (LumpSumTax.class.equals(customer.getEntrepreneurshipForm().taxPaymentForm().getClass())) {
+            BigDecimal monthlyRevenue = countMonthlyRevenueService.countMonthlyRevenue(
+                            customer.getCustomerId(),
+                            monthToCalculateHealthInsuranceContribution
+                    )
+                    .amount();
+
+            //do 5k
+            if (monthlyRevenue.compareTo(BigDecimal.valueOf(FIRST_CONTRIBUTION_RANGE)) < 0) {
+                return new Money(
+                        monthlyRevenue
+                                .multiply(BigDecimal.valueOf(0.6f))
+                                .multiply(BigDecimal.valueOf(0.09f))
+                        , Currency.PLN);
+            }
+            // 5k - 25k
+            else if (
+                    monthlyRevenue.compareTo(BigDecimal.valueOf(FIRST_CONTRIBUTION_RANGE)) >= 0 &&
+                            monthlyRevenue.compareTo(BigDecimal.valueOf(SECOND_CONTRIBUTION_RANGE)) < 0
+            ) {
+                return new Money(monthlyRevenue.multiply(BigDecimal.valueOf(0.09)), Currency.PLN);
+            }
+            //powyzej 25k
+            else {
+                return new Money(
+                        monthlyRevenue
+                                .multiply(BigDecimal.valueOf(1.8))
+                                .multiply(BigDecimal.valueOf(0.09)),
+                        Currency.PLN);
+            }
+
+        }
+
+        if (FlatTax.class.equals(customer.getEntrepreneurshipForm().taxPaymentForm().getClass())) {
+            Money monthlyIncome = countIncomeService.countMonthlyIncome(
+                    customer,
+                    monthToCalculateHealthInsuranceContribution
+            );
+            if (monthlyIncome.amount().compareTo(MINIMUM_WAGE) <= 0) {
+                return new Money(MINIMUM_WAGE.multiply(BigDecimal.valueOf(0.09)), Currency.PLN);
+
+            } else {
+                return new Money(monthlyIncome.amount().multiply(BigDecimal.valueOf(0.49)), Currency.PLN);
+            }
+
+        }
+
+
+        if (GeneralTax.class.equals(customer.getEntrepreneurshipForm().taxPaymentForm().getClass())) {
+            Money monthlyIncome = countIncomeService.countMonthlyIncome(
+                    customer,
+                    monthToCalculateHealthInsuranceContribution
+            );
+            if (monthlyIncome.amount().compareTo(MINIMUM_WAGE) <= 0) {
+                return new Money(MINIMUM_WAGE.multiply(BigDecimal.valueOf(0.09)), Currency.PLN);
+
+            } else {
+                return new Money(monthlyIncome.amount().multiply(BigDecimal.valueOf(0.09)), Currency.PLN);
+            }
+
+        }
+
+
+
+        return null;
```

Comment:
null?

Reply:
Thought that the method should never come to this point. Wrong assumption -> provided throwing new Exception at this point, with logging before the throwing.
Resolved